ugs-pendant/src/main/java/com/willwinder/universalgcodesender/pendantui/v1/resources/MachineResource.java

```diff
@@ -0,0 +1,199 @@
+package com.willwinder.universalgcodesender.pendantui.v1.resources;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import com.willwinder.universalgcodesender.IController;
+import com.willwinder.universalgcodesender.connection.ConnectionDriver;
+import com.willwinder.universalgcodesender.connection.ConnectionFactory;
+import com.willwinder.universalgcodesender.model.Axis;
+import com.willwinder.universalgcodesender.model.BackendAPI;
+import com.willwinder.universalgcodesender.model.BaudRateEnum;
+import com.willwinder.universalgcodesender.pendantui.v1.model.GcodeCommands;
+import com.willwinder.universalgcodesender.services.JogService;
+import com.willwinder.universalgcodesender.utils.FirmwareUtils;
+import com.willwinder.universalgcodesender.utils.Settings;
+import com.willwinder.universalgcodesender.utils.SettingsFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.NotAcceptableException;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import java.io.BufferedReader;
+import java.io.StringReader;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Path("/v1/machine")
+public class MachineResource {
+
+    @Inject
+    private BackendAPI backendAPI;
+
+    @Inject
+    private JogService jogService;
+
+    @GET
+    @Path("connect")
+    @Produces(MediaType.APPLICATION_JSON)
+    public void connect() throws Exception {
+        if (backendAPI.isConnected()) {
+            throw new NotAcceptableException("Already connected");
+        }
+        Settings settings = SettingsFactory.loadSettings();
+        backendAPI.connect(settings.getFirmwareVersion(), settings.getPort(), Integer.valueOf(settings.getPortRate()));
+    }
+
+    @GET
+    @Path("disconnect")
+    @Produces(MediaType.APPLICATION_JSON)
+    public void disconnect() throws Exception {
+        if (!backendAPI.isConnected()) {
+            throw new NotAcceptableException("Not connected");
```

Comment:
This can probably be a no-op rather than an error.

Reply:
I'll leave it as it is for now. This is getting fixed in the candy mechanics branch.